cmd/kubectl/kubectlcobra/run.go

```diff
@@ -0,0 +1,140 @@
+package kubectlcobra
+
+import (
+	"context"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/kubectl/pkg/cmd/apply"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/kstatus/wait"
+	"time"
+)
+
+func newApplier(factory util.Factory, ioStreams genericclioptions.IOStreams) *Applier {
+	return &Applier{
+		applyOptions: apply.NewApplyOptions(ioStreams),
+		statusOptions: &StatusOptions{},
+		factory: factory,
+		ioStreams: ioStreams,
+	}
+}
+
+type Applier struct {
+	factory util.Factory
+	ioStreams genericclioptions.IOStreams
+
+	applyOptions *apply.ApplyOptions
+	statusOptions *StatusOptions
+	resolver *wait.Resolver
+}
+
+func (a *Applier) Initialize(cmd *cobra.Command) error {
+	a.applyOptions.PreProcessorFn = PrependGroupingObject(a.applyOptions)
+	err := a.applyOptions.Complete(a.factory, cmd)
+	if err != nil {
+		return err
+	}
+
+	resolver, err := a.newResolver(2 * time.Second)
```

Comment:
what is this time?  it should at least be a constant -- maybe this should be configurable via flag ?

Reply:
Updated this to set this value through the flag.